### PR TITLE
Fixes #12390 - Sending requests to proxy as well when deleting arf re…

### PR DESCRIPTION
**Pull request by xprazak2**
…ports

### CODE REVIEW

**Comment by shlomizadok** [comment]
[test]

File: app/models/foreman_openscap/arf_report.rb [hunk 1]
```diff
@@ -154,6 +154,14 @@ def equal?(other)
         policy.id == other.policy.id
     end
 
+    def destroy
+      if proxy.destroy_report(self, ForemanOpenscap::Helper::find_name_or_uuid_by_host(host))
+        super
+      else
+        false
```
Comment (shlomizadok) [inline app/models/foreman_openscap/arf_report.rb]: Maybe add a log error that it failed cause it failed on proxy?

**Comment by shlomizadok** [comment]
@xprazak2 - please note that tests are failing.

File: app/lib/proxy_api/openscap.rb [hunk 2]
```diff
@@ -36,5 +36,15 @@ def arf_report_bzip(report, cname)
...
       end
     end
+
+    def destroy_report(report, cname)
+      begin
+        parse(delete("arf/#{report.id}/#{cname}/#{report.reported_at.to_i}/#{report.policy_arf_report.digest}"))
+      rescue => e
+        logger.error "Failed to destroy arf report with id #{report.id} on Smart Proxy"
+        logger.debug e.backtrace.join("\n\t")
+        false
```
Comment (xprazak2) [inline app/lib/proxy_api/openscap.rb]: We are already logging when there is a failure on the proxy.

**Comment by shlomizadok** [comment]
merging. Thanks @xprazak2 !

**Pull request merged**